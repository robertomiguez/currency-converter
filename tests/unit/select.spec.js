import { shallowMount } from '@vue/test-utils'
import Select from '@/components/Select.vue'

let wrapper

beforeEach(() => {
  wrapper = shallowMount(Select, {
    propsData: {
      options: ['BTC', 'ETH', 'DASH', 'BAT']
    },
    mocks: {},
    stubs: {},
    methods: {}
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('Select', () => {
  test('testing component props', () => {
    expect(wrapper.find('.select-css').text().replace(/\s+/g, ' '))
      .toBe('BTC ETH DASH BAT')
  })

  test('regression test', () => {
    expect(wrapper.element)
      .toMatchSnapshot()
  })
})
