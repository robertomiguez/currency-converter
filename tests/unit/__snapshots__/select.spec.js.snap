// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select regression test 1`] = `
<select
  class="select-css"
>
  <option
    value="BTC"
  >
    
        BTC
    
  </option>
  <option
    value="ETH"
  >
    
        ETH
    
  </option>
  <option
    value="DASH"
  >
    
        DASH
    
  </option>
  <option
    value="BAT"
  >
    
        BAT
    
  </option>
</select>
`;
